fetch/fetch_test.go

```diff
@@ -312,7 +313,7 @@ func TestFetch_RegisterPeerHashes(t *testing.T) {
 func TestFetch_PeerDroppedWhenMessageResultsInValidationReject(t *testing.T) {
 	lg := logtest.New(t)
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
-	t.Cleanup(cancel)
+	defer cancel()
```

Comment:
What's the difference here? When to use one or the other?

Reply:
Order of execution - `defer`s are executed before cleanups:

```golang
func Test_Defer(t *testing.T) {
	t.Cleanup(func() { t.Log("cleanup") })
	defer t.Log("deferred")
	t.Cleanup(func() { t.Log("cleanup2") })
	defer t.Log("deferred2")
}
```

prints:

```
=== RUN   Test_Defer
    fetch_test.go:409: deferred2
    fetch_test.go:409: deferred
    fetch_test.go:407: cleanup2
    fetch_test.go:405: cleanup
--- PASS: Test_Defer (0.00s)
PASS
```